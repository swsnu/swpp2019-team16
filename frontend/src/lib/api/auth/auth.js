import client from '../client';

export function login({ email, password }) {
  return client.post('/api/v1/user/login', { email, password });
}

export function register({ userType, email, password, carType, plateNo }) {
  return client.post('/api/v1/user/register', {
    userType,
    email,
    password,
    carType,
    plateNo,
  });
}
