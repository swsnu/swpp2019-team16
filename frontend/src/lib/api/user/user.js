import client from '../client';

export function get({ id }) {
  return client.get(`/api/v1/user/${id}`);
}

export function updatePoint({ userId, point }) {
  return client.put(`/api/v1/user/${userId}`, { point });
}

export function riderOnTaxi({ riderId }) {
  return client.put(`/api/v1/user/taxi/rider/${riderId}`);
}

export function driverGoTaxi({ driverId }) {
  return client.put(`/api/v1/user/taxi/driver/${driverId}`);
}
