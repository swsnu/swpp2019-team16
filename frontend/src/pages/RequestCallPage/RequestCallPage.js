import React from 'react';

function RequestCallPage() {
  return <div>RequestCall Page!</div>;
}
export default RequestCallPage;
