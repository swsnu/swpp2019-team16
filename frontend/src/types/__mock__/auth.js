export const mockRiderAuth = {
  id: 1,
  email: 'rider1@gmail.com',
  user_type: 'rider',
  vehicle: {
    car_type: '',
    plate_no: '',
  },
};

export const mockDriverAuth = {
  id: 1,
  email: 'driver1@gmail.com',
  user_type: 'driver',
  vehicle: {
    car_type: '',
    plate_no: '',
  },
};
