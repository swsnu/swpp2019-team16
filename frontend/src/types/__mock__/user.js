export const mockRider = {
  id: 1, // rider/driver id
  status: 'IDLE',
  group: 1, // group id
  user: {
    id: 2,
    email: 'rider1@gmail.com',
    user_type: 'rider',
    vehicle: {
      car_type: '',
      plate_no: '',
    },
  },
};

export const mockDriver = {
  id: 1, // rider/driver id
  status: 'IDLE',
  group: 1, // group id
  user: {
    id: 2,
    email: 'driver1@gmail.com',
    user_type: 'driver',
    vehicle: {
      car_type: 'benz',
      plate_no: '1234',
    },
  },
};

export const createMockRider = (id, groupId, email) => {
  return {
    id,
    status: 'IDLE',
    group: groupId,
    user: {
      id: 2,
      email,
      user_type: 'rider',
      vehicle: {
        car_type: '',
        plate_no: '',
      },
    },
  };
};

export const createMockDriver = (id, groupId, email) => {
  return {
    id,
    status: 'IDLE',
    group: groupId,
    user: {
      id: 2,
      email,
      user_type: 'driver',
      vehicle: {
        car_type: 'benz',
        plate_no: '1234',
      },
    },
  };
};
