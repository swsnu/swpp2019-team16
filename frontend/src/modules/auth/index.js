import auth from './auth';

export * from './auth';
export default auth;
