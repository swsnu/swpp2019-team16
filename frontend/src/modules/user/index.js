import user from './user';

export * from './user';
export default user;
