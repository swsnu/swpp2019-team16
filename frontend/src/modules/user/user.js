import { createAction, handleActions } from 'redux-actions';
import { takeLatest, put } from 'redux-saga/effects';
import * as userAPI from '../../lib/api/user';
import createRequestSaga, {
  createRequestActionTypes,
} from '../../lib/createRequestSaga';

const TEMP_SET_USER = 'user/TEMP_SET_USER';
const SOMEONE_ON_TAXI = 'user/SOMEONE_ON_TAXI';
const GO_TAXI = 'user/GO_TAXI';
const INIT_POINT_UPDATED_FLAG = 'user/INIT_POINT_UPDATED_FLAG';
const [CHECK, CHECK_SUCCESS, CHECK_FAILURE] = createRequestActionTypes(
  'user/CHECK',
);
const [LOGOUT, LOGOUT_SUCCESS, LOGOUT_FAILURE] = createRequestActionTypes(
  'user/LOGOUT',
);
const [
  UPDATE_POINT,
  UPDATE_POINT_SUCCESS,
  UPDATE_POINT_FAILURE,
] = createRequestActionTypes('user/UPDATE_POINT');
const [
  RIDER_ON_TAXI,
  RIDER_ON_TAXI_SUCCESS,
  RIDER_ON_TAXI_FAILURE,
] = createRequestActionTypes('user/RIDER_ON_TAXI');
const [
  DRIVER_GO_TAXI,
  DRIVER_GO_TAXI_SUCCESS,
  DRIVER_GO_TAXI_FAILURE,
] = createRequestActionTypes('user/DRIVER_GO_TAXI');

export const tempSetUser = createAction(TEMP_SET_USER, user => user);
export const check = createAction(CHECK, ({ id }) => ({ id }));
export const logout = createAction(LOGOUT);
export const updatePoint = createAction(UPDATE_POINT, ({ userId, point }) => ({
  userId,
  point,
}));

export const checkSaga = createRequestSaga(CHECK, userAPI.get);
export const riderOnTaxi = createAction(RIDER_ON_TAXI, ({ riderId }) => ({
  riderId,
}));
export const goTaxi = createAction(DRIVER_GO_TAXI, ({ driverId }) => ({
  driverId,
}));
export const someoneOnTaxi = createAction(SOMEONE_ON_TAXI, ({ riderId }) => ({
  riderId,
}));
export const taxiGone = createAction(GO_TAXI);
export const initPointUpdatedFlag = createAction(INIT_POINT_UPDATED_FLAG);

const riderOnTaxiSaga = createRequestSaga(RIDER_ON_TAXI, userAPI.riderOnTaxi);
const driverGoTaxiSaga = createRequestSaga(
  DRIVER_GO_TAXI,
  userAPI.driverGoTaxi,
);

function checkFailureSaga() {
  try {
    localStorage.removeItem('user');
  } catch (e) {
    console.error('localStorage is not working');
  }
}

export function* logoutSaga() {
  try {
    localStorage.removeItem('user');
    yield put({ type: LOGOUT_SUCCESS });
  } catch (e) {
    console.error(e);
    yield put({ type: LOGOUT_FAILURE });
  }
}

export const updatePointSaga = createRequestSaga(
  UPDATE_POINT,
  userAPI.updatePoint,
);
export function* userSaga() {
  yield takeLatest(CHECK, checkSaga);
  yield takeLatest(CHECK_FAILURE, checkFailureSaga);
  yield takeLatest(LOGOUT, logoutSaga);
  yield takeLatest(UPDATE_POINT, updatePointSaga);
  yield takeLatest(RIDER_ON_TAXI, riderOnTaxiSaga);
  yield takeLatest(DRIVER_GO_TAXI, driverGoTaxiSaga);
}

const initialState = {
  user: null,
  checkError: null,
  logoutError: null,
  updatePointError: null,
  onTaxiRidersList: [],
  driverGoTaxi: false,
  onTaxiError: null,
  goTaxiError: null,
  pointUpdated: false,
};

const user = handleActions(
  {
    [TEMP_SET_USER]: (state, { payload: user }) => ({
      ...state,
      user,
    }),
    [CHECK_SUCCESS]: (state, { payload: user }) => ({
      ...state,
      user,
      checkError: null,
    }),
    [CHECK_FAILURE]: (state, { payload: error }) => ({
      ...state,
      user: null,
      checkError: error,
    }),
    [LOGOUT_SUCCESS]: state => ({
      ...state,
      user: null,
      logoutError: null,
    }),
    [LOGOUT_FAILURE]: (state, { payload: error }) => ({
      ...state,
      logoutError: error,
    }),
    [UPDATE_POINT_SUCCESS]: (state, { payload: user }) => ({
      ...state,
      user: {
        ...user,
        user,
      },
      pointUpdated: true,
      updatePointError: null,
    }),
    [UPDATE_POINT_FAILURE]: (state, { payload: error }) => ({
      ...state,
      updatePointError: error,
    }),
    [RIDER_ON_TAXI_FAILURE]: (state, { payload: error }) => ({
      ...state,
      onTaxiError: error,
    }),
    [DRIVER_GO_TAXI_FAILURE]: (state, { payload: error }) => ({
      ...state,
      goTaxiError: error,
    }),
    [SOMEONE_ON_TAXI]: (state, { payload: riderId }) => ({
      ...state,
      onTaxiRidersList: [...state.onTaxiRidersList, riderId],
    }),
    [GO_TAXI]: state => ({
      ...state,
      driverGoTaxi: true,
    }),
    [INIT_POINT_UPDATED_FLAG]: state => ({
      ...state,
      pointUpdated: initialState.pointUpdated,
    }),
  },
  initialState,
);

export default user;
