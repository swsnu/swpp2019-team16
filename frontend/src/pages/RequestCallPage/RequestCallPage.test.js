import React from 'react';
import RequestCallPage from './RequestCallPage';
import { renderWithRedux } from '../../test/utils';
import { MemoryRouter, Route } from 'react-router-dom';
import { mockRider } from '../../types/__mock__/user';
import { mockRiderAuth } from '../../types/__mock__/auth';

describe('<RequestCallPage />', () => {
  it('SHOULD match with snapshot', async () => {
    const state = {
      user: {
        user: mockRider,
      },
      group: {
        group: null,
      },
      auth: {
        auth: mockRiderAuth,
      },
    };

    const { container } = renderWithRedux(
      <MemoryRouter initialEntries={['/requestcall']}>
        <Route component={RequestCallPage} path={'/requestcall'} />
      </MemoryRouter>,
      state,
    );
    expect(container).toMatchSnapshot();
  });
});
