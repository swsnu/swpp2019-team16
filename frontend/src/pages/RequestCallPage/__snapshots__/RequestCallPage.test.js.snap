// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestCallPage /> SHOULD match with snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa bPGLpq"
  >
    <div
      class="makeStyles-root-1"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h4
          class="MuiTypography-root makeStyles-title-3 MuiTypography-h4"
        >
          Ya-Ta!
        </h4>
        <div>
          <button
            aria-controls="menu-appbar"
            aria-haspopup="true"
            aria-label="account of current user"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="makeStyles-root-63"
    >
      <div
        class="makeStyles-title-64"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2"
          title="We are waiting for carpool group"
        >
          We are waiting for carpool group
        </h2>
      </div>
      <div>
        <div>
          <label
            class="MuiFormControlLabel-root"
            data-testid="STT Mode"
          >
            <span
              aria-disabled="false"
              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-80 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
            >
              <span
                class="MuiIconButton-label"
              >
                <input
                  class="PrivateSwitchBase-input-83"
                  data-indeterminate="false"
                  name="STT"
                  type="checkbox"
                  value="STT Mode"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              STT Mode
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
</div>
`;
